impedancefitter/fitter.py

```diff
@@ -134,10 +135,10 @@ class Fitter(object):
         Exists only when :meth:`run` was called.
     """
 
-    def __init__(self, inputformat, directory=None, **kwargs):
+    def __init__(self, inputformat, directory=None, df=None, **kwargs):
```

Comment:
I would suggest to use only `DF` as the new inputformat.
Then, add the information that you had in the `inputformat` string as kwargs:
Instead of `DF Freq-Re1-Im1` use `inputformat=DF`, `df_freq_column=Freq`, `df_re_column=Re`, `df_im_column=Im`. The limitation for the user would be to make sure that the columns have the same pattern and are numbered with a clear indexing (e.g., 1-indexed as in your example).

Would that suit your purpose?